### PR TITLE
Use hawkular-commons 0.9.1, update CORS

### PR DESCRIPTION
**Pull request by jotak**
It means splitting cors filtering into a request filter and a response filter.
Validation is processed during request, adding headers dureing response.

### CODE REVIEW

**Comment by objectiser** [comment]
@jpkrohling Could you review this one please?

**Comment by jmazzitelli** [comment]
Once @jpkrohling reviews and this gets merged, I'll rebase my PR #718

File: server/rest/src/main/java/org/hawkular/apm/server/rest/filter/CorsRequestFilter.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    @PostConstruct
+    public void initAllowedOrigins() {
+        String allowedCorsOrigins = PropertyUtil.getProperty(PropertyUtil.HAWKULAR_APM_CORS_ALLOWED_ORIGINS);
+        originValidation = new OriginPredicate(
+                allowedCorsOrigins == null ? Headers.ALLOW_ALL_ORIGIN : allowedCorsOrigins);
+    }
+
+    @Override public void filter(ContainerRequestContext requestContext) throws IOException {
+        CorsFilters.filterRequest(requestContext, originValidation);
```
Comment (jpkrohling) [inline server/rest/src/main/java/org/hawkular/apm/server/rest/filter/CorsRequestFilter.java]: Why do we need the filter defined here, if all the logic is on commons? Isn't it possible to have the filter on a JAR within the WAR? The only code that is specific to APM is the `originValidation`, but I assume every module has a similar code, so, even that can be reused, no?

Comment (jotak) [inline server/rest/src/main/java/org/hawkular/apm/server/rest/filter/CorsRequestFilter.java]: Actually, there are some slight differences in inventory, metrics, apm and alerts: not on "filter", but in the way to provide "extraAllowedHeaders" and "allowedCorsOrigins".
They could of course fit in an abstracted model, with the abstract class calling abstract methods but I think relying here on composition offers better readability and decreases coupling (ie. allows to use another API for filtering, for instance)

**Comment by jmazzitelli** [comment]
We need this PR merged before the okhttp3 PR #718 is merged. Can we get this merged asap?

**Pull request merged**